services/cloudwatch/src/it/java/software/amazon/awssdk/services/cloudwatch/CloudWatchIntegrationTest.java

```diff
@@ -164,6 +167,87 @@ public void put_get_metricdata_list_metric_returns_success() throws
         assertTrue(seenDimensions);
     }
 
+    /**
+     * Tests putting metrics with request compression and then getting them back.
+     * TODO: We can remove this test once CloudWatch adds "RequestCompression" trait to PutMetricData
+     */
+    @Test
+    public void put_get_metricdata_list_metric_withRequestCompression_returns_success() {
```

Comment:
Can we have Junit test cases something like this Test Class https://github.com/aws/aws-sdk-java-v2/blob/81982a83b5267c60e88ab50c8e656a0a56c8024f/test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/SyncHttpChecksumInTrailerTest.java#L158-L185 which will cover error cases and also Streaming and non streaming operation 


And here you can verify that the data which you send to Service is as expected by doing 

```java
        verify(putRequestedFor(anyUrl()).withRequestBody(
            containing(
                expectedCompressedRequestBody)));
```